vision_datasets/common/data_manifest/data_manifest.py

```diff
@@ -199,3 +199,82 @@ def __eq__(self, other) -> bool:
 
     def __len__(self):
         return len(self.images)
+
+
```

Comment:
Nit: Should we add the annotation-level class before the definition for `DatasetManifest` for better organization of annotation vs dataset level classes?